src/Kraken/Config.hs

```diff
@@ -22,13 +23,42 @@ defaultKrakenConfig = KrakenConfig {
   numberOfRetries = 1
  }
 
-loadConfig :: FilePath -> IO (KrakenConfig, Config)
-loadConfig configFile = do
-  config <- load [Required configFile]
-  let custom = subconfig "customConfig" config
-  result <- KrakenConfig <$>
-     lookup config "retryDelay" <*>
-     lookupDefault 1 config "numberOfRetries"
+newtype ConfigPair a = ConfigPair { _unConfigPair :: (KrakenConfig, a) }
+                     deriving (Eq, Show)
+
+instance FromJSON a => FromJSON (ConfigPair a) where
+  parseJSON (Object v) = do
+    delay <- v .: "retryDelay"
+    retries <- v .:? "numberOfRetries" >>= return . fromMaybe 1
+    custom <- v .: "customConfig"
+    return $ ConfigPair (KrakenConfig delay retries, custom)
+
+  parseJSON _ = fail "FromJSON.parseJSON failure for ConfigPair"
+
+withConfig :: FromJSON a => FilePath -> (a -> IO b) -> IO b
+withConfig p f = BS.readFile p >>= return . decodeEither >>= \case
+  Right config -> f config
+  Left m -> throwIO . ErrorCall $
+            "Kraken.Config.withConfig: decode failure ‘" ++ m ++ "’"
```

Comment:
If we just throw an exception in the case of parse errors, we could have something of type `FromJSON a => FilePath -> IO a`, right?